### PR TITLE
[LLDB][NFC] Remove dead code

### PR DESCRIPTION
**Pull request by walter-erquinigo**
I found some type/typesystem code that is dead and some of it seems to have been replaced by the ValueObjectPrinter.

### CODE REVIEW

File: lldb/include/lldb/Symbol/TypeSystem.h [hunk 1]
```diff
@@ -384,13 +384,15 @@ class TypeSystem : public PluginInterface,
...
-                         bool show_summary, bool verbose, uint32_t depth) = 0;
+  /*
+    virtual void DumpValue(lldb::opaque_compiler_type_t type,
+                           ExecutionContext *exe_ctx, Stream &s,
+                           lldb::Format format, const DataExtractor &data,
+                           lldb::offset_t data_offset, size_t data_byte_size,
+                           uint32_t bitfield_bit_size,
+                           uint32_t bitfield_bit_offset, bool show_types,
+                           bool show_summary, bool verbose, uint32_t depth) = 0;
+                           */
```
Comment (clayborg) [inline lldb/include/lldb/Symbol/TypeSystem.h:396]: remove this, don't just comment it out?

Comment (walter-erquinigo) [reply lldb/include/lldb/Symbol/TypeSystem.h:396]: I had updated the patch where I effectively removed it. Sorry, it seems you saw the oldest version.

Comment (clayborg) [reply lldb/include/lldb/Symbol/TypeSystem.h:396]: It is still here even with "show all commits"?

Comment (walter-erquinigo) [reply lldb/include/lldb/Symbol/TypeSystem.h]: lol, I had stashed the change where I delete the code. 
@clayborg , now this patch should be fine

**Review by clayborg — COMMENTED** [review]
Looks fine to me, but anyone at Apple might need to comment to make sure it doesn't take away something needed for Swift.

File: lldb/include/lldb/Symbol/CompilerType.h [hunk 2]
```diff
@@ -183,8 +181,6 @@ class CompilerType {
 
   bool IsPointerToScalarType() const;
 
-  bool IsRuntimeGeneratedType() const;
```
Comment (jimingham) [inline lldb/include/lldb/Symbol/CompilerType.h:186]: Swift uses this one.

**Comment by jimingham** [comment]
Swift uses the `IsRuntimeGeneratedType` API, so it would be nice if you could leave that in to reduce diffs.  All the others are not used in the swift support.

**Comment by walter-erquinigo** [comment]
@jimingham , thanks! I reverted the changes to that method and left a tiny comment mentioning that it's used by swift.

**Comment by walter-erquinigo** [comment]
Thanks, @JDevlieghere !

**Comment by walter-erquinigo** [comment]
@clayborg PTAL

**Review by bulbazord — APPROVED** [review]
Thanks for taking the time to clean this up, and thanks for taking the time to make sure this wouldn't hurt us downstream with swift!

**Comment by walter-erquinigo** [comment]
@bulbazord , thanks!! I'll try to improve the type system and other parts of the LLDB as I add support for my language.

**Pull request merged**